kernel/aster-nix/src/syscall/semop.rs

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::time::Duration;
+
+use super::SyscallReturn;
+use crate::{
+    ipc::semaphore::system_v::{
+        sem::{sem_op, SemBuf},
+        sem_set::{check_sem, SEMOPM},
+        PermissionMode,
+    },
+    prelude::*,
+    time::timespec_t,
+};
+
+pub fn sys_semop(sem_id: i32, tsops: Vaddr, nsops: usize, _ctx: &Context) -> Result<SyscallReturn> {
+    debug!(
+        "[sys_semop] sem_id = {:?}, tsops_vaddr = {:x?}, nsops = {:?}",
+        sem_id, tsops, nsops
+    );
+    do_sys_semtimedop(sem_id, tsops, nsops, None)
+}
+
+pub fn sys_semtimedop(
+    sem_id: i32,
+    tsops: Vaddr,
+    nsops: usize,
+    timeout: Vaddr,
+    _ctx: &Context,
+) -> Result<SyscallReturn> {
+    debug!(
+        "[sys_semtimedop] sem_id = {:?}, tsops_vaddr = {:x?}, nsops = {:?}, timeout_vaddr = {:x?}",
+        sem_id, tsops, nsops, timeout
+    );
+
+    let timeout = if timeout == 0 {
+        None
+    } else {
+        Some(Duration::try_from(
+            CurrentUserSpace::get().read_val::<timespec_t>(timeout)?,
```

Comment:
Instead of `CurrentUserSpace::get()`, you can call `ctx.get_user_space`, which can avoid calling `Task::current`.